File: src/api/AMOGUS.Core/Domain/Models/Entities/GameSession.cs
using AMOGUS.Core.Domain.Enums;
using AMOGUS.Infrastructure.Identity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AMOGUS.Core.Domain.Models.Entities {
    public class GameSession {

        [Key]
        public string SessionId { get; set; } = string.Empty;

        [ForeignKey("User")]
        public string UserId { get; set; } = string.Empty;

        public ApplicationUser? User { get; set; }

        public double Playtime { get; set; }

        public int CorrectAnswersCount { get; set; }

        public int GivenAnswersCount { get; set; }

        public double AverageTimePerQuestion { get; set; }

        public double QuickestAnswer { get; set; }

        public double SlowestAnswer { get; set; }

        public CategoryType Category { get; set; }

        public DateTime PlayedAt { get; set; }

        [NotMapped]
        public List<Question> Questions { get; set; } = new();

    }
}

File: src/ui/src/app/core/interfaces/game-session.ts
import { question } from "./question";
export class GameSession {

    sessionId : string;
    userId : string;
    playTime: number;
    correctAnswersCount: number;
    givenAnswersCount: number;

    averageTimePerQuestion: number;
    quickestAnswer: number;
    slowestAnswer: number;

    category: CategoryType;

    questions: question[]

    constructor(sessionId: string, userId: string, playTime: number, correctAnswersCount: number, givenAnswersCount: number, averageTimePerQuestion: number, quickestAnswer:number, slowestAnswer: number, category: CategoryType, questions: question[]) {
        this.sessionId = sessionId;
        this.userId = userId;
        this.playTime = playTime;
        this.correctAnswersCount = correctAnswersCount;
        this.givenAnswersCount = givenAnswersCount;
        this.averageTimePerQuestion = averageTimePerQuestion;
        this.quickestAnswer = quickestAnswer;
        this.slowestAnswer = slowestAnswer;
        this.category = category;
        this.questions = questions;
    }
}

export enum CategoryType {
    MENTAL,
    ANALYSIS,
    GEOMETRY,
    RANDOMMENTAL,
    RANDOMMENTAL_INSANE
}

